src/table.js

```diff
@@ -810,52 +810,51 @@ function getAllKeys(rows) {
   return Array.from(keys);
 }
 
-export function inferSchema(source) {
-  const allKeys = getAllKeys(source);
+export function inferSchema(source, columns = getAllKeys(source)) {
   const schema = [];
   const sampleSize = 100;
   let sample = source.slice(0, sampleSize);
   if (arrayIsPrimitive(sample)) {
     sample = sample.map((d) => {
       return {value: d};
     });
-    allKeys.push("value");
+    columns.push("value");
```

Comment:
(This is now mutating an input. Perhaps we can move the promotion of primitive arrays to {value} objects to before we infer the schema, so that inferSchema doesn’t have to handle primitive arrays.)